### PR TITLE
Html render function

### PR DESCRIPTION
**Pull request by scottsfarley93**
Exposes a render method that allows custom html to be displayed in the dropdown menu.

### CODE REVIEW

File: src/suggestions.js [hunk 1]
```diff
@@ -22,76 +22,74 @@ var Suggestions = function(el, data, options) {
 
   this.list.draw();
 
-  this.el.addEventListener('keyup', function(e) {
+  this.el.addEventListener('keyup', function (e) {
```
Comment (tristen) [inline src/suggestions.js]: Can you remove the paren formatting throughout this pr?

File: src/suggestions.js [hunk 2]
```diff
@@ -212,8 +204,26 @@ Suggestions.prototype.getCandidates = function(callback) {
...
+ * @return {String} html
+ */
+Suggestions.prototype.render = function (item) {
+  var boldString = this.getItemValue(item);
+  var indexString = this.normalize(boldString);
+
+  var indexOfQuery = indexString.lastIndexOf(this.query);
+  while (indexOfQuery > -1) {
+    var endIndexOfQuery = indexOfQuery + this.query.length;
+    boldString = boldString.slice(0, indexOfQuery) + '<strong>' + boldString.slice(indexOfQuery, endIndexOfQuery) + '</strong>' + boldString.slice(endIndexOfQuery);
```
Comment (tristen) [inline src/suggestions.js:241]: Not sure I'm following this. if this `render` method allows a user to template a list item, would wrapping `<strong>` be desired or expected?

**Review by tristen — COMMENTED** [review]
Sweet! Just left some initial comments.

File: src/suggestions.js (hunk 2 above)
Comment (scottsfarley93) [reply src/suggestions.js:241]: If the user provides a custom `render` method, then that will be used instead. If the user does not provide a custom render function, then this default function will be used.  The default function follows the existing functionality of wrapping the matching text in `<strong>` tags.

Comment (tristen) [reply src/suggestions.js:241]: oh dear me 🙈 I see now the existing functionality in the diff. 👍

File: src/suggestions.js [hunk 3]
```diff
@@ -66,8 +69,8 @@ Suggestions.prototype.handleKeyDown = function(e) {
     case 13: // ENTER
     case 9: // TAB
       if (!this.list.isEmpty()) {
-        if (this.list.isVisible()) {
-          e.preventDefault();
+        if (this.list.isVisible()) {
```
Comment (tristen) [inline src/suggestions.js]: nit: seeing some extra whitespace here and the line below

**Review by tristen — COMMENTED** [review]
Overall, the changes here LGTM. Running the demo locally, I'm seeing a method error on the string passed to fuzzy:

Screen Shot 2019-03-20 at 12 55 27 PM

**Review by tristen — APPROVED** [review]
💘

**Pull request merged**